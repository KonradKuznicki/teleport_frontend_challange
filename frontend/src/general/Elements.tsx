import React from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { Link } from 'react-router-dom';

export const theme = {
    font: "'M PLUS 2', sans-serif",
    fontWeight: 400,
    fontWeightBolder: 500,
    contentMasWidth: '1000px',
    colors: {
        lighter1: '#181A23',
        primary1: '#08170C',
        darker1: '#000000',
        primary2: '#FFD764',
        lighter2: '#ffe7bc',
        lightest2: '#ffffff',
        primary3: '#afffb8',
        darker3: '#82fe8f',
        primary4: '#005de0',
        darker4: '#0010be',
        danger: '#ff001f',
        careful: '#ff6a00',
        good: '#57ff63',
        info: '#00ffff',
    },
};

createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=M+PLUS+2:wght@800&display=swap');
  @import url('https://fonts.googleapis.com/icon?family=Material+Icons'); // TODO: why it doesn't work?

  html, body {
    height: 100%;
    min-height: 100%;

    margin: 0;
    padding: 0;

    font-family: ${theme.font};
  }

`;

// font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;

export const Title = styled.h1`
    font-size: 30px;
    text-align: center;
    font-family: ${(props) => props.theme.font};
    font-weight: ${(props) => props.theme.fontWeightBolder};

    margin: 0;
    padding: 0;
`;

export const Box = styled.div`
    box-sizing: border-box;
    padding: 20px;
    background: ${(props) => props.theme.colors.primary2};
`;

export const Label = styled.label`
    font-family: ${(props) => props.theme.font};
    font-weight: ${(props) => props.theme.defaultFontWeight};
`;

export const Input = styled.input`
    border: none;
    border-bottom: 2px solid ${(props) => props.theme.colors.darker1};
    padding: 5px;
    font-size: 17px;
    box-sizing: border-box;
    background: ${(props) => props.theme.colors.lighter2};

    &:focus {
        outline: none;
        background: ${(props) => props.theme.colors.lightest2};
    }
`;

export const A = styled(Link)`
    color: ${(props) => props.theme.colors.primary4};
    font-family: ${(props) => props.theme.font};
    font-weight: ${(props) => props.theme.defaultFontWeight};
    text-decoration: underline;

    :hover {
        // text-decoration: underline;
        color: ${(props) => props.theme.colors.darker4};
    }
`;

export const Button = styled.button`
    border: none;
    background: ${(props) => props.theme.colors.primary3};
    padding: 10px 20px;
    font-size: 20px;
    cursor: pointer;

    :hover {
        background: ${(props) => props.theme.colors.darker3};
    }
`;

export const Row = styled.div`
    margin-top: 17px;
    margin-bottom: 17px;
`;

export const Cell = styled.div`
    display: inline-block;
    // width: 47.5%;
    width: 50%;

    > Input {
        width: 100%;
    }
`;

export const Page = styled.div`
    font-family: ${(props) => props.theme.font};
    background-color: ${(props) => props.theme.colors.lighter1};
    // padding: 5px 100px 100px 100px;
    font-size: 20px;
    height: 100vh;
    width: 100vw;
    display: table-cell;
    vertical-align: middle;
`;

export const Icon = styled.span`
    font-family: 'Material Icons', emoji;
    font-weight: normal;
    font-style: normal;
    font-size: 30px; /* Preferred icon size */
    display: inline-block;
    line-height: 1;
    text-transform: none;
    letter-spacing: normal;
    word-wrap: normal;
    white-space: nowrap;
    direction: ltr;
    vertical-align: middle;

    /* Support for all WebKit browsers. */
    -webkit-font-smoothing: antialiased;
    /* Support for Safari and Chrome. */
    text-rendering: optimizeLegibility;

    /* Support for Firefox. */
    -moz-osx-font-smoothing: grayscale;

    /* Support for IE. */
    font-feature-settings: 'liga';
`;

export function CustomIcon({ code, ...props }: { code: string }) {
    return (
        <Icon {...props}>
            <span className="material-icons-sharp">{code}</span>
        </Icon>
    );
}

export const DeleteIcon = (props: JSX.IntrinsicAttributes) => (
    <CustomIcon code="delete_outline" {...props} />
);
export const DownloadIcon = () => <CustomIcon code="file_download" />;
