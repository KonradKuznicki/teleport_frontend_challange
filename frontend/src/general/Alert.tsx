import React from 'react';
import styled from 'styled-components';
import { CustomIcon } from './Elements';


const Closer = styled(CustomIcon).attrs({ code: 'close' })`
  font-size: 50px;
  display: inline-block;
  vertical-align: middle;
  margin-right: 20px;
`;

const AlertContainer = styled.div`
  text-align: center;
  padding: 40px;
`;

const AlertContent = styled.div`
  font-family: ${(props) => props.theme.font};
  font-size: 30px;
  display: inline;
  vertical-align: middle;
`;

function AlertStructure(props: any) {

    return <AlertContainer {...props} >
        <Closer />
        <AlertContent>{props.children}</AlertContent>
    </AlertContainer>;
}

export const Alert = styled(AlertStructure)`
  color: ${props => props.theme.colors[props.type]};
  width: 100%;
  display: block;
  max-width: ${(props) => props.theme.contentMasWidth};
  margin: auto;
  cursor: pointer;

`;