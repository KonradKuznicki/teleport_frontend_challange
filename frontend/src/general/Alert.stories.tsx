import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { ThemeProvider } from 'styled-components';
import { Page, theme } from './Elements';
import { Alert } from './Alert';


export default {
    title: 'general/Alerts',
    component: Alert,
} as ComponentMeta<typeof Alert>;

const Template: ComponentStory<typeof Alert> = (args) => <ThemeProvider theme={theme}><Page><Alert
    type={args.type}>{args.content}</Alert></Page></ThemeProvider>;

export const Default = Template.bind({});
export const Error = Template.bind({});
export const Waring = Template.bind({});
export const Success = Template.bind({});


const content = 'quick brown fox something something...';

Default.args = { content, type: 'info' };
Error.args = { content, type: 'danger' };
Waring.args = { content, type: 'careful' };
Success.args = { content, type: 'good' };
