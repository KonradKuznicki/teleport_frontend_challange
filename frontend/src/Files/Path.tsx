import React from 'react';
import styled from 'styled-components';
import { A } from '../general/Elements';

const List = styled.ul`
    margin: 0;
    padding: 0;
`;

const Item = styled.li`
    display: inline-block;
    margin: 5px;
`;

type PathPart = {
    name: string;
    path: string;
};

const DefaultRoot: PathPart[] = [{ name: 'My Files', path: '/' }];

export function Path({ parts, ...props }: { parts: string[] }) {
    const items = DefaultRoot.concat(
        parts.map((i, c, a) => ({
            name: i,
            path: (c ? a[c - 1] : '') + '/' + i,
        })),
    );

    const parents = items.slice(0, -1);
    const current = items[items.length - 1];

    const pathPart = ({ name, path }: PathPart) => (
        <>
            <Item key={name + 'i'}>
                <A key={name + 'a'} to={'/files' + path}>
                    {name}
                </A>
            </Item>
            <Item key={name + '-slash'}>/</Item>
        </>
    );

    return (
        <List {...props} key="ul">
            {parents.map(pathPart)}
            <Item key="current">{current.name}</Item>
        </List>
    );
}
