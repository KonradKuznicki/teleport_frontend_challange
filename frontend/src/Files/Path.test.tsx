import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { Path } from './Path';
import { ThemeProvider } from 'styled-components';
import { theme } from '../general/Elements';

describe('Path', function () {
    describe('empty path', function () {
        const root = 'My Files';

        let node: RenderResult<
            typeof import('@testing-library/dom/types/queries'),
            HTMLElement
        >;
        beforeEach(() => {
            node = render(
                <ThemeProvider theme={theme}>
                    <Path parts={[]} />
                </ThemeProvider>,
            );
        });
        test('shows root', function () {
            expect(node.getByText(root)).toBeInTheDocument();
        });
        test('no slash when only root', function () {
            expect(node.queryAllByText('/')).toHaveLength(0);
        });
        it('should match snapshot', function () {
            expect(node.asFragment().firstChild).toMatchSnapshot();
        });
    });

    describe('non empty path', function () {
        const root = 'My Files';

        let node: RenderResult<
            typeof import('@testing-library/dom/types/queries'),
            HTMLElement
        >;
        beforeEach(() => {
            node = render(
                <ThemeProvider theme={theme}>
                    <Path parts={['documents', 'images']} />
                </ThemeProvider>,
            );
        });
        test('shows root', function () {
            expect(node.getByText(root)).toBeInTheDocument();
        });
        test('shows slashes between items', function () {
            expect(node.queryAllByText('/')).toHaveLength(2);
        });
        it('should match snapshot', function () {
            expect(node.asFragment().firstChild).toMatchSnapshot();
        });
    });
});
