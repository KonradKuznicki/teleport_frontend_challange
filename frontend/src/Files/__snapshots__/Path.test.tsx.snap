// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Path empty path should match snapshot 1`] = `
<ul
  class="sc-jrQzAO jVOqxz"
>
  <li
    class="sc-kDTinF fyeoYR"
  >
    My Files
  </li>
</ul>
`;

exports[`Path non empty path should match snapshot 1`] = `
<ul
  class="sc-jrQzAO jVOqxz"
>
  <li
    class="sc-kDTinF fyeoYR"
  >
    <a
      class="sc-eCImPb cpDrvT"
      href="/My Files"
    >
      My Files
    </a>
  </li>
  <li
    class="sc-kDTinF fyeoYR"
  >
    /
  </li>
  <li
    class="sc-kDTinF fyeoYR"
  >
    <a
      class="sc-eCImPb cpDrvT"
      href="/documents"
    >
      documents
    </a>
  </li>
  <li
    class="sc-kDTinF fyeoYR"
  >
    /
  </li>
  <li
    class="sc-kDTinF fyeoYR"
  >
    images
  </li>
</ul>
`;
